vortex-serde/src/layouts/read/recordbatchreader.rs

```diff
@@ -0,0 +1,60 @@
+use std::sync::Arc;
+
+use arrow_array::{RecordBatch, RecordBatchReader};
+use arrow_schema::{ArrowError, SchemaRef};
+use futures::StreamExt;
+use vortex::arrow::infer_schema;
+use vortex::Array;
+use vortex_error::{VortexError, VortexResult};
+
+use super::LayoutBatchStream;
+use crate::io::VortexReadAt;
+
+fn vortex_to_arrow_error(error: VortexError) -> ArrowError {
+    ArrowError::ExternalError(Box::new(error))
+}
+
+fn vortex_to_arrow(result: VortexResult<Array>) -> Result<RecordBatch, ArrowError> {
+    result
+        .and_then(RecordBatch::try_from)
+        .map_err(vortex_to_arrow_error)
+}
+
+pub struct VortexRecordBatchReader<R: VortexReadAt + Unpin + Send + 'static> {
+    stream: LayoutBatchStream<R>,
+    arrow_schema: SchemaRef,
+    runtime: tokio::runtime::Runtime,
+}
+
+impl<R: VortexReadAt + Unpin + Send + 'static> VortexRecordBatchReader<R> {
```

Comment:
fwiw this really only needs Unpin and 'static you can kill the Send bound all throughout

Reply:
Hmm. Indeed. I got pretty confused at one point trying to create a `Box<dyn RecordBatchReader + Send>`. Removed.